app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,65 @@
+class CommentsController < ApplicationController
+
+  def new
+    @comment = Comment.new
+  end
+
+  def create
+    comment = resource.comments.build(comment_params.merge(user: current_user))
+
+    if comment.save
+      flash[:success] = "Comment is created."
+      redirect_to redirect_url_after_create resource
```

Comment:
If we did the commentable thing we wouldn't need the second call to resource here.